torchinductor/lowering.py

```diff
@@ -1076,6 +1312,29 @@ def fn(idx):
         # else: torch.exp(torch.log(a) * b)
 
 
+def mutate_to(changed, val):
```

Comment:
Is TorchInductor aiming to support in-place ops in full generality?

Reply:
So if we functionalized things you just wouldn't have these lowerings/

Reply:
There are some things like input mutation which I don't think functionalization will handle.  Some decomposition are introducing mutation also.  So even if you functionalize before AOT autograd, you get graphs with mutation in them.

Maybe we can delete these at some point.  Though I think they will be needed for a while.

Reply:
> Some decomposition are introducing mutation also. So even if you functionalize before AOT autograd, you get graphs with mutation in them.

Well, we'd presumably functionalize during AOTAutograd tracing, right? So it'd go through decompositions.

I do agree with input mutation... not totally sure how to handle that.